tests/s3/versioning/test_delete_object_tagging.py

```diff
@@ -348,3 +343,175 @@ def test_del_obj_tag_40436(self):
+                    self.object_name, latest_v)
+        resp = self.s3_ver_obj.get_obj_tag_ver(bucket_name=self.bucket_name,
+                                               object_name=self.object_name, version=latest_v)
+        assert_utils.assert_true(resp[0], resp)
+        # For null version ID, expecting "TagSet": []
+        assert_utils.assert_false(resp[1], resp)
+        LOGGER.info("Step 7: Performed GET Object Tagging for %s with versionId=%s",
+                    self.object_name, latest_v)
+
+        LOGGER.info("Step 8: Perform Delete Object Tagging for %s with versionId specified %s",
+                    self.object_name, latest_v)
+        resp = self.s3_ver_obj.delete_obj_tag_ver(bucket_name=self.bucket_name,
+                                                  object_name=self.object_name,
+                                                  version=latest_v)
+        assert_utils.assert_true(resp[0], resp)
+        # For null version ID, expecting "TagSet": []
+        assert_utils.assert_false(resp[1], resp)
+        LOGGER.info("Step 8: Performed Delete Object Tagging for %s with versionId specified",
+                    self.object_name)
+
+        LOGGER.info("ENDED:Test DELETE object tagging for a deleted versioned object")
+
+    @pytest.mark.s3_ops
+    @pytest.mark.tags("TEST-40439")
+    @CTFailOn(error_handler)
+    def test_del_obj_tag_40439(self):
+        """
+        Test DELETE object tagging for non-existing version or object.
+        """
+        LOGGER.info(
+            "Test DELETE object tagging for non-existing version or object.")
+        self.ver_tag.update({self.object_name: {}})
+        LOGGER.info("Step 1: Perform PUT Bucket versioning with status as Enabled on %s",
+                    self.bucket_name)
+        resp = self.s3_ver_obj.put_bucket_versioning(bucket_name=self.bucket_name)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 1: Performed PUT Bucket versioning with status as Enabled on %s",
+                    self.bucket_name)
+
+        LOGGER.info("Step 2: Upload object %s to create version on bucket %s",
+                    self.object_name, self.bucket_name)
+        s3_ver_tlib.upload_version(self.s3_test_obj,
+                                   bucket_name=self.bucket_name,
+                                   file_path=self.file_path1,
+                                   object_name=self.object_name,
+                                   versions_dict=self.versions)
+        latest_v = self.versions[self.object_name]["version_history"][-1]
+        LOGGER.info("Step 2: Successfully uploaded object %s enabling versioning on "
+                    "bucket %s with version ID %s", self.object_name, self.bucket_name, latest_v)
+
+        LOGGER.info("Step 3: Perform PUT Object Tagging for %s with a tag key-value pair",
+                    self.object_name)
+        resp = s3_ver_tlib.put_object_tagging(s3_tag_test_obj=self.s3_tag_obj,
```

Comment:
Please update the log message and the put_object_tagging step to include version_id set to the object's latest version id

Reply:
Done, however I can see same can be fetched from versions_dict if  not specified.